Riot/Modules/Onboarding/OnboardingCoordinatorBridgePresenter.swift

```diff
@@ -0,0 +1,146 @@
+// File created from FlowTemplate
+// $ createRootCoordinator.sh Onboarding/SplashScreen Onboarding OnboardingSplashScreen
+/*
+ Copyright 2021 New Vector Ltd
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import Foundation
+
+@objcMembers
+class OnboardingCoordinatorBridgePresenterParameters: NSObject {
+    /// The external registration parameters for AuthenticationViewController.
+    var externalRegistrationParameters: [AnyHashable: Any]?
+    /// The credentials to use after a soft logout has taken place.
+    var softLogoutCredentials: MXCredentials?
+}
+
+/// OnboardingCoordinatorBridgePresenter enables to start OnboardingCoordinator from a view controller.
+/// This bridge is used while waiting for global usage of coordinator pattern.
+/// **WARNING**: This class breaks the Coordinator abstraction and it has been introduced for **Objective-C compatibility only** (mainly for integration in legacy view controllers). Each bridge should be removed once the underlying Coordinator has been integrated by another Coordinator.
+@objcMembers
+final class OnboardingCoordinatorBridgePresenter: NSObject {
```

Comment:
This is a temporary bridge presenter to switch in the onboarding flow inside of `MasterTabBarController` for now. The plan is to remove this as create `OnboardingCoordinator` inside of `TabBarCoordinator`, however I still need to look into how `LegacyAppDelegate` should communicates with the tab bar coordinator so will do this in a later PR.